packages/jest-environment-jsdom/src/index.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {Script} from 'vm';
+import {Global, Config} from '@jest/types';
+import mock, {ModuleMocker} from 'jest-mock';
+import {installCommonGlobals} from 'jest-util';
+import {JestFakeTimers as FakeTimers} from '@jest/fake-timers';
+import {JestEnvironment, EnvironmentContext} from '@jest/environment';
+import {JSDOM, VirtualConsole} from 'jsdom';
+
+// The `Window` interface does not have an `Error.stackTraceLimit` property, but
+// `JSDOMEnvironment` assumes it is there.
+interface Win extends Window {
+  Error: {
+    stackTraceLimit: number;
+  };
+}
+
+function isWin(globals: Win | Global.Global): globals is Win {
+  return (globals as Win).document !== undefined;
+}
+
+class JSDOMEnvironment implements JestEnvironment {
+  dom: JSDOM | null;
+  fakeTimers: FakeTimers<number> | null;
+  // @ts-ignore
+  global: Global.Global | Win | null;
+  errorEventListener: ((event: Event & {error: unknown}) => void) | null;
+  moduleMocker: ModuleMocker | null;
+
+  constructor(config: Config.ProjectConfig, options: EnvironmentContext = {}) {
+    this.dom = new JSDOM('<!DOCTYPE html>', {
+      pretendToBeVisual: true,
+      runScripts: 'dangerously',
+      url: config.testURL,
+      virtualConsole: new VirtualConsole().sendTo(options.console || console),
+      ...config.testEnvironmentOptions,
+    });
+
+    // `defaultView` returns a `Window` type, which is missing the
+    // `Error.stackTraceLimit` property. See the `Win` interface above.
+    const global = (this.global = this.dom.window.document
+      .defaultView as Win | null);
+
+    if (!global || !this.global) {
```

Comment:
what is this check for? Doesn't seem like it could ever happen

Reply:
`this.dom.window.document.defaultView` is `Window | null` so when I did the type assertion to `Win` I wanted to replicate the behavior (that null is a possible value). 

We can always override it, if we are 100% sure `defaultView` is never null. But their typings say it may be null.

Reply:
Eg. the check is needed with both the default type of `defaultView`, as well as with my overridden type of `defaultView`. But if we change it as you suggest, then the check should also be removed.

Reply:
`global` can never be `null`, `this.global` might be `null` (since we set it nu `null` ourselves in `teardown`).

>We can always override it, if we are 100% sure `defaultView` is never null. But their typings say it may be null.

I don't think we've ever gotten a report that this is `null`. I'd rather do something like this:

```ts
const global = this.dom.window.document.defaultView;

if (global === null) {
  // better error
  throw new Error('wat')
}

this.global = global;
```

But I think saying `as Window` is enough - as mentioned I've never heard of anyone having issues with this being `null`

Reply:
Updated